store/store.go

```diff
@@ -112,6 +112,79 @@ type Store struct {
 	treeStoreLockDir string
 }
 
+func (s *Store) UpdateSize(key string, newSize int64) error {
+	return s.db.Do(func(tx *sql.Tx) error {
+		_, err := tx.Exec("UPDATE aciinfo SET size = $1 WHERE blobkey == $2", newSize, key)
+		return err
+	})
+}
+
+func (s *Store) UpdateTreeStoreSize(key string, newSize int64) error {
+	return s.db.Do(func(tx *sql.Tx) error {
+		_, err := tx.Exec("UPDATE aciinfo SET treestoresize = $1 WHERE blobkey == $2", newSize, key)
+		return err
+	})
+}
+
+func (s *Store) populateSize() error {
+	var ais []*ACIInfo
+	err := s.db.Do(func(tx *sql.Tx) error {
+		var err error
+		ais, err = GetACIInfosWithKeyPrefix(tx, "")
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("populateSize(): error retrieving ACI Infos: %v", err)
+	}
+
+	aciSizes := make(map[string]int64)
+	tsSizes := make(map[string]int64)
+	for _, ai := range ais {
+		key := ai.BlobKey
+
+		im, err := s.ReadStream(key)
+		if err != nil {
+			return err
+		}
+
+		rd, err := io.Copy(ioutil.Discard, im)
+		if err != nil {
+			return err
+		}
+		aciSizes[key] = int64(rd)
+
+		id, err := s.GetTreeStoreID(key)
+		if err != nil {
+			return err
+		}
+
+		tsR, err := s.treestore.IsRendered(id)
+		if err != nil {
+			return fmt.Errorf("error determining whether the tree store is rendered: %v", err)
+		}
+		if tsR {
+			tsSize, err := s.treestore.Size(id)
+			if err != nil {
+				return err
+			}
+
+			treepath := s.treestore.GetPath(id)
```

Comment:
This block looks rather unrelated to populating the size. Would you mind adding a comment what is this for?